src/AI/CapPhaseAnalyzer.java

```diff
@@ -0,0 +1,454 @@
+      {
+        sb.append(String.format("  chain\n"));
+        for( CapStop stop : chain )
+          sb.append(String.format("    %s\n", stop));
+      }
+    }
+    return sb.toString();
+  }
+
+  /**
+   * If the input unit has a cap chain allocated, return it
+   * Else, if it's on the start of a cap chain in the un-allocated list, remove the chain from the list and allocate it
+   * @return The chosen chain, if any
+   */
+  public ArrayList<CapStop> getCapChain(Unit unit)
+  {
+    ArrayList<CapStop> chain = null;
+    XYCoord position = new XYCoord(unit);
+    // Add the unit to a cap chain, if possible
+    if( capChainsAllocated.containsKey(unit) )
+      chain = capChainsAllocated.get(unit);
+    else
+    {
+      ArrayList<ArrayList<CapStop>> chainList = capChains.get(position);
+      if( null != chainList && !chainList.isEmpty() )
+      {
+        chain = chainList.remove(0);
+        capChainsAllocated.put(unit, chain);
+        if( chainList.isEmpty() )
+          capChains.remove(position);
+      }
+    }
+
+    return chain;
+  }
+
+  public GameAction getCapAction(GameMap map, Unit unit)
+  {
+    XYCoord position = new XYCoord(unit);
+    ArrayList<CapStop> chain = getCapChain(unit);
+    if( null != chain )
+    {
+      if( unit.getCaptureProgress() > 0 )
+      {
+        return new CaptureLifecycle.CaptureAction(map, unit, Utils.findShortestPath(unit, position, map));
+      }
+
+      while (!chain.isEmpty())
+      {
+        final XYCoord coord = chain.get(0).coord;
+        // If we're already there or own it, throw the point out
+        if( position.equals(coord) || !unit.CO.isEnemy(map.getLocation(coord).getOwner()) )
+          chain.remove(0);
+        else
+          return findChainAction(map, unit, chain);
+      }
+      // Chain's done, if we ever get here
+      capChainsAllocated.remove(unit);
+    }
+
+    return null;
+  }
+
+  public GameAction findChainAction(GameMap gameMap, Unit unit, ArrayList<CapStop> chain)
+  {
+    XYCoord goal = chain.get(0).coord;
+
+    boolean includeOccupiedSpaces = false; // We assume that the area won't be cluttered
+    ArrayList<XYCoord> destinations = Utils.findPossibleDestinations(unit, gameMap, includeOccupiedSpaces);
+
+    // If we can get to our destination, go for it
+    if( destinations.contains(goal) )
+      return new CaptureLifecycle.CaptureAction(gameMap, unit, Utils.findShortestPath(unit, goal, gameMap));
+
+    Utils.sortLocationsByTravelTimeToCoord(unit, destinations, gameMap, goal);
+
+    for( XYCoord moveCoord : destinations )
+    {
+      Unit resident = gameMap.getLocation(moveCoord).getResident();
+      boolean spaceFree = null == resident;
+      if( !spaceFree )
+        continue; // I don't know how we got here, but bail
+
+      // Figure out how to get here.
+      GamePath movePath = Utils.findShortestPath(unit, moveCoord, gameMap);
+
+      return new WaitLifecycle.WaitAction(unit, movePath);
+    }
+    return null;
```

Comment:
Instead of all this, why not just call `AIUtils.moveTowardLocation()`?

Reply:
...because I keep forgetting it exists?
Yeah, that sounds about right.

Reply:
... how?
"I need a way to move a unit towards some a location"
"Too bad that use case hasn't come up yet. Pretty sure we don't do that at all"
"Ah well, easy enough to add it."

Reply:
Realistically, this is copied Wally code that doesn't use that function because it's integrated with the eviction loop and thus needs more insight into the details.
This code doesn't have anything like that since we assume there aren't units in the way when you're doing your initial capture runs.

I could conceive of maybe making a variant of the function that ignores friendlies so the caller can decide where to evict and/or snip.